lib/Backends/NNPI/NNPIMLTraceWrapper.cpp

```diff
@@ -291,11 +306,14 @@ bool NNPITraceContext::createInternalContext(bool swTraces, bool hwTraces) {
     }
   }
   if (hwTraces) {
+    uint64_t maxBufferSize = hardwareBufferSizeMB > 0
+                                 ? hardwareBufferSizeMB
+                                 : MAX_HW_TRACE_BUFFER_SIZE_MB;
     IceCapsHwTraceConfig traceConfigs[2];
     traceConfigs[0].traceOptions.config_type =
         eIceCapsHwTraceConfigType::ICE_CAPS_HWTRACE_OPTIONS;
     traceConfigs[0].traceOptions.device_mask = devMask_;
-    traceConfigs[0].traceOptions.max_trace_size = MAX_TRACE_BUFFER_SIZE;
+    traceConfigs[0].traceOptions.max_trace_size = maxBufferSize;
```

Comment:
Not necessary here to multiply by `1024 * 1024`, as is done for `max_bytes`?

Reply:
The max_trace_size for the hardware traces is in mega bytes already - no need to multiply.

Reply:
Perhaps add a comment here specifying this is intentional, since this is different than `max_bytes` so looks like it could be a bug because they're not consistent.

Reply:
Added comments to both places.